### PR TITLE
Several small changes to laplace_transform

### PR DESCRIPTION
**Pull request by hanspi42**
#### References to other Issues or PRs
Psrt of #24561

#### Brief description of what is fixed or changed
This PR contains several small changes made while checking for the completeness of the laplace_transform rules.  Changes:
- wrong gamma function in simple rules corrected
- better treatment of Heaviside(b-t) for positive b
- rule for t**np*g for positive integer np implemented (Bateman 4.1 (6))

#### Release Notes

- integrals
  - several small changes and error corrections in laplace_transform

### CODE REVIEW

File: sympy/integrals/laplace.py [hunk 1]
```diff
@@ -576,6 +577,15 @@ def _laplace_rule_heaviside(f, t, s):
...
             return (r, pr, cr)
+        ma2 = ma1[y].match(a-t)
+        if ma2 and ma2[a].is_positive:
+            _debug('     rule: Heaviside window open')
+            r, pr, cr = _laplace_transform(
+                (1-Heaviside(t-ma2[a]))*ma1[g], t, s, simplify=False)
+            return (r, pr, cr)
+        if ma2 and ma2[a].is_negative:
+            _debug('     rule: Heaviside window closed')
+            return (0, 0, S.true)
```
Comment (smichr) [inline sympy/integrals/laplace.py:588]:
```suggestion
        if ma2 := ma1[y].match(a - t):
            if ma2[a].is_positive:
                _debug('     rule: Heaviside window open')
                r, pr, cr = _laplace_transform(
                    (1 - Heaviside(t - ma2[a]))*ma1[g], t, s, simplify=False)
                return (r, pr, cr)
            if ma2[a].is_negative:
                _debug('     rule: Heaviside window closed')
                return (0, 0, S.true)
```

Comment (hanspi42) [reply sympy/integrals/laplace.py:588]: Nice, thanks! I will also alter my other change in this PR accordingly. Long live the Walrus operator ;)

**Comment by hanspi42** [comment]
Hi all,

I have not found more things to change in my manual inspection of the code base, so this pull request can be merged now.

I'll re-run the checks now.

**Comment by oscarbenjamin** [comment]
The authors check failed I think because this branch is quite old compared to master now: a merge or rebase is needed.

Otherwise the changes look good to me.

**Comment by hanspi42** [comment]
Thanks! I merged it and have enabled auto-merge.

**Pull request merged**